### PR TITLE
[Intel MKL]: Upgrading to MKL-DNN v0.15

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR upgrades Intel MKL-DNN library to v0.15. It also includes a small change that makes convolution compatible with that library version.

### CODE REVIEW

**Review by andydavis1 — DISMISSED** [review]

**Comment by mahmoud-abuzaina** [comment]
@caisq these unsuccessful tests don't seem to be related to this PR. They are failing for others PR too.

**Comment by tatianashp** [comment]
+@gunan Bazel file changes MKL-DNN file from .tar.gz to .zip. Will this work on Linux?

File: tensorflow/core/util/mkl_util.h [hunk 1]
```diff
@@ -1883,9 +1883,9 @@ class MklPrimitive {
  public:
   virtual ~MklPrimitive() {}
 
-  // Dummy data. Its size, hard-coded as 256 here, does
-  // not matter since MKL should never operate on this buffer.
-  unsigned char DummyData[256];
+  // Dummy data.
+  // does not matter since MKL should never operate on this buffer.
```
Comment (tatianashp) [inline tensorflow/core/util/mkl_util.h]: Please fix the comment to be a complete meaningful sentence.

**Review by tatianashp — CHANGES_REQUESTED** [review]

**Comment by gunan** [comment]
In theory, zip files also should work, but I would still keep the tar.gz files to be sure.

**Comment by tatianashp** [comment]
@mahmoud-abuzaina Could you use tar.gz for MKL-DNN? Also, please fix the comment.

**Comment by mahmoud-abuzaina** [comment]
@tatianashp Both are fixed.

**Comment by mahmoud-abuzaina** [comment]
it seems there is a conflict introduced few minutes ago because of some change on the master branch. Rebasing now...

**Comment by mahmoud-abuzaina** [comment]
Rebase is done.

**Review by tatianashp — APPROVED** [review]

**Comment by mahmoud-abuzaina** [comment]
@gunan is this ready to be merged?

**Comment by gunan** [comment]
It is in the process of merging internally.

**Pull request merged**